### PR TITLE
Write non-breaking space as "~" in LaTeXWriter

### PR DESCRIPTION
**Pull request by goerz**
In DocumenterCitations, non-breaking spaces (`nbsp="\u00A0"`) are used pretty extensively, e.g., in author names (`"M.$(nbsp)H.$(nbsp)Goerz"`).

Currently, the `LaTeXWriter` leaves these unchanged, so that a unicode non-breaking space ends up in the TeX file. Depending on fonts installed on the user's system and their general LaTeX setup, that doesn't tend to work too well, see https://github.com/JuliaDocs/DocumenterCitations.jl/pull/46#issuecomment-1761021964.

This PR recognizes `\u00A0` as a character that needs to be "escaped" and replaces it with the standard `~` that LaTeX traditionally uses to indicate non-breaking spaces.

### CODE REVIEW

File: Project.toml
Comment (mortenpi) [inline Project.toml:3]:
```suggestion
version = "1.1.2"
```

File: CHANGELOG.md
Comment (mortenpi) [inline CHANGELOG.md:6]:
```suggestion
## Version [v1.1.2] - 2023-10-23
```

File: CHANGELOG.md
Comment (mortenpi) [inline CHANGELOG.md:10]:
```suggestion
* Non-breaking spaces are now properly converted as "~" in the `LaTeXWriter`. ([#2300])
```

**Review by mortenpi — APPROVED** [review]
LGTM, thanks @goerz! Will tag as soon as this is merged.

**Comment by mortenpi** [comment]
@goerz It looks like the maintainer access to the PR is not enabled. Do you mind enabling it so that I could push a small additional test?

**Comment by goerz** [comment]
> Do you mind enabling [maintainer access]

I would certainly not mind… but I don't seem to be able to figure out how!

**Comment by mortenpi** [comment]
From a different PR, but there should be a thingy like this at the bottom of the right sidebar:

image

And it should also ask that under the PR description text box when you open a PR.

**Comment by goerz** [comment]
I'm pretty sure I've seen that "Allow edits and access to secrets by maintainers" option before on other PRs I've opened, but with Documenter, it's just not there for me, for this PR or for https://github.com/JuliaDocs/Documenter.jl/pull/2330 which I just opened. I have absolutely no idea why. In any case, I've given you write access to my entire [goerz-forks/Documenter.jl repo](https://github.com/goerz-forks/Documenter.jl) from which I'm opening these PRs, so hopefully that'll be sufficient.

**Comment by goerz** [comment]
Ah! It's a Github bug! https://github.com/orgs/community/discussions/5634

I have a `goerz-forks` organization that I normally use for making PRs so as not to litter my main account with cloned throw-away forks. Maybe I'll have to rethink that :-/

**Pull request merged**
